src/metrics/server.rs

```diff
@@ -73,12 +73,26 @@ async fn handle_metrics(
             .expect("builder with known status code should not fail");
     }
 
+    let response_content_type = match req
+        .headers()
+        .get(http::header::ACCEPT)
+        .unwrap_or(&http::HeaderValue::from_str("").expect("static str must parse"))
+        .to_str()
+        .unwrap_or_default()
+        .to_lowercase()
+        .split(";")
+        .collect::<Vec<_>>()
+        .first()
```

Comment:
Do we know that prometheus will always scrape with openmetrics as the first 'accept'

Reply:
Good point, that is likely not a safe invariant

Reply:
Finally got back around to this. Got it reworked to support multiple Accept headers and added testing.